fs/_is_subdir.ts

```diff
@@ -19,7 +20,7 @@ export function isSubdir(
   dest: string | URL,
   sep = SEPARATOR,
 ): boolean {
-  if (src === dest) {
+  if (isSamePath(src, dest)) {
```

Comment:
`isSamePath` internally calls `toPathString` for `src` and `dest`, which is also called for `src` and `dest` in `isSubdir`. I think we should avoid the these duplicated calls

Reply:
Perhaps we can use `resolve()` directly instead of using `isSamePath()`:

```ts
export function isSubdir(
  src: string | URL,
  dest: string | URL,
  sep = SEPARATOR,
): boolean {
  src = toPathString(src);
  dest = toPathString(dest);

  if (resolve(src) === resolve(dest)) {
    return false;
  }

  const srcArray = src.split(sep);
  const destArray = dest.split(sep);
  return srcArray.every((current, i) => destArray[i] === current);
}
```

Reply:
@babiabeo Looks good to me. Could you make a PR with a test case?

Reply:
Cool. I'll close this PR then 👍🏾